clicommand/agent_start.go

```diff
@@ -180,10 +180,11 @@ type AgentStartConfig struct {
 	TracingPropagateTraceparent bool   `cli:"tracing-propagate-traceparent"`
 
 	// Other shared flags
-	StrictSingleHooks         bool   `cli:"strict-single-hooks"`
-	KubernetesExec            bool   `cli:"kubernetes-exec"`
-	TraceContextEncoding      string `cli:"trace-context-encoding"`
-	NoMultipartArtifactUpload bool   `cli:"no-multipart-artifact-upload"`
+	StrictSingleHooks                  bool   `cli:"strict-single-hooks"`
+	KubernetesExec                     bool   `cli:"kubernetes-exec"`
+	KubernetesLogCollectionGracePeriod string `cli:"kubernetes-log-collection-grace-period"`
```

Comment:
This should "just work" if its type was `time.Duration`. I'm not sure why some other duration flags go via a string and then manually parse it. The config loader supports `time.Duration`. 

```suggestion
	KubernetesLogCollectionGracePeriod time.Duration `cli:"kubernetes-log-collection-grace-period"`
```